cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/ProposeNewConstitution.hs

```diff
@@ -319,19 +314,41 @@ hprop_ledger_events_propose_new_constitution = H.integrationWorkspace "propose-n
 
   void $ evalEither eConstitutionAdopted
 
-foldBlocksCheckProposalWasSubmitted
-  :: TxId -- TxId of submitted tx
-  -> Env
-  -> LedgerState
-  -> [LedgerEvent]
-  -> BlockInMode -- Block i
-  -> Maybe LedgerEvent -- ^ Accumulator at block i - 1
-  -> IO (Maybe LedgerEvent, FoldStatus) -- ^ Accumulator at block i and fold status
-foldBlocksCheckProposalWasSubmitted txid _ _ allEvents _ _ = do
-  let newGovProposal = filter (filterNewGovProposals txid) allEvents
-  if null newGovProposal
-  then return (Nothing, ContinueFold)
-  else return (Just $ newGovProposal !! 0, StopFold)
+  finalGovState <- H.note $ work </> gov </> "final_gov_state.json"
+
+  void $ H.execCli' execConfig
+    [ "conway", "query", "gov-state"
+    , "--volatile-tip"
+    , "--out-file", finalGovState
+    ]
+
+  -- Tally registered votes
+
+  finalGovFileBS <- liftIO $ LBS.readFile finalGovState
+
+  votes <- H.nothingFail $ do (Aeson.Object jsonValue) <- Aeson.decode finalGovFileBS
+                              (Aeson.Array proposals) <- jsonValue KeyMap.!? "proposals"
+                              (Aeson.Object proposal) <- Vector.headM proposals
+                              (Aeson.Object votes) <- proposal KeyMap.!? "dRepVotes"
+                              KeyMap.foldl' extractVote (Just []) votes
+
+  length (filter (== "VoteYes") votes) === 4
+  length (filter (== "VoteNo") votes) === 3
+  length (filter (== "Abstain") votes) === 2
+  length votes === numVotes
+
+  where
+    extractVote :: Maybe [String] -> Aeson.Value -> Maybe [String]
+    extractVote (Just acc) (Aeson.String x) = Just $ unpack x:acc
+    extractVote _ _ = Nothing
```

Comment:
Any value which isn't 'String' here will short-circuit this foldl making it return `Nothing`. Is that expected?
I think `Either` with an error message would be better here, it's hard to figure out why the failure is happening here at the first glance.

Reply:
The solution to the [previews comment](https://github.com/IntersectMBO/cardano-node/pull/5731#discussion_r1544534861) already made `extractVote` unnecessary so I removed it altogether (see [this commit](https://github.com/IntersectMBO/cardano-node/commit/f4089571337428fba2ee0efe632dade014c0d862))